src/features/SensorStatus/SensorStatusManager.ts

```diff
@@ -1,59 +1,79 @@
 import { SensorStatus } from './SensorStatusSlice';
 import { DatabaseService } from '../../common/services';
 
-const SENSOR_STATUS = `
-with breach as (
-  select (select count(*) > 0
-  from temperaturebreach tb
-  where tb.acknowledged = 0 and sensorid = ?
-  and temperaturebreachconfigurationid = "HOT_BREACH") hasHotBreach,
-  (select count(*) > 0
-  from temperaturebreach tb
-  where tb.acknowledged = 0 and sensorid = ?
-  and temperaturebreachconfigurationid = "COLD_BREACH") hasColdBreach
-  )
-  
-  SELECT s.id id, (select hasHotBreach from breach) hasHotBreach, (select hasColdBreach from breach) hasColdBreach,
-  s.macAddress macAddress,
-  s.logInterval logInterval,
-  s.name        name,
-  s.logDelay    logDelay,
-  mostRecentLogTimestamp,
-  s.batteryLevel < 25 isLowBattery,
-  firstTimestamp,
-  coalesce(numberOfLogs,0) > 0 hasLogs,
-  coalesce(numberOfLogs,0) numberOfLogs,
-  currentTemperature,
-  CASE when mostRecentLogTimestamp - firstTimestamp > (3 * 24 * 60 * 60) then mostRecentLogTimestamp - 24 * 3 * 60 * 60 else firstTimestamp end as minChartTimestamp,
-  CASE
-  WHEN endTimestamp IS NULL AND temperatureBreachConfigurationId = 'HOT_BREACH' THEN 1 ELSE 0 END AS isInHotBreach,
-  CASE WHEN endTimestamp IS NULL AND temperatureBreachConfigurationId = 'COLD_BREACH' THEN 1 ELSE 0 END AS isInColdBreach
-  FROM      sensor s OUTER
-  LEFT JOIN (SELECT coalesce(min(timestamp), 0) firstTimestamp from temperaturelog where sensorid = ?)
-  left JOIN
-  (
-    SELECT  max(timestamp) mostRecentLogTimestamp,
-            temperature    currentTemperature,
-            count(*) numberOfLogs,
-            tl.sensorid,
-            *
-    FROM temperaturelog tl 
-    LEFT OUTER JOIN
-      (
-        SELECT   max(startTimestamp) startTimestamp,
-        temperatureBreachConfigurationId,
-        endTimestamp,
-        sensorid
-        FROM temperaturebreach tb
-        GROUP BY sensorid
-  
-      ) tb
-    ON tl.sensorid = tb.sensorid
-    GROUP BY tl.sensorid
-    order by timestamp
-  ) logs
-  ON logs.sensorid = s.id 
-  where s.id = ?
+enum SensorStatusQueryKey {
+  CurrentTemperature = 'currentTemperature',
+  MostRecentLogTimestamp = 'mostRecentLogTimestamp',
+  NumberOfLogs = 'numberOfLogs',
+  FirstTimestamp = 'firstTimestamp',
+  NumberOfHotBreaches = 'numberOfHotBreaches',
+  NumberOfColdBreaches = 'numberOfColdBreaches',
+  batteryLevel = 'batteryLevel',
+}
+
+type SensorStatusQueryResult = {
+  [SensorStatusQueryKey.CurrentTemperature]: number;
+  [SensorStatusQueryKey.FirstTimestamp]: number;
+  [SensorStatusQueryKey.MostRecentLogTimestamp]: number;
+  [SensorStatusQueryKey.NumberOfColdBreaches]: number;
+  [SensorStatusQueryKey.NumberOfHotBreaches]: number;
+  [SensorStatusQueryKey.NumberOfLogs]: number;
+  [SensorStatusQueryKey.batteryLevel]: number;
+};
+
+const buildSensorStatusQuery = (whereClause = 'id = ?') => `
```

Comment:
Tried to tidy this up. I am not sure right now about queries.

- Keep them in the same file is easier
- Keep them in a seperate file is a bit cleaner
- Keep them in a separate sql file is more cleaner (with syntax highlighting) but requires to read them in as a string 🤷